### ISSUE TITLE
DL4J Spark: fix layerwise pretraining via spark

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#6656]
A couple of test failures:
TestSparkMultiLayerParameterAveraging
* testVaePretrainSimple
* testVaePretrainSimpleCG

We may need new methods for this (given we changed from pretrain/backprop to dedicated pretrainLayer methods).

Also I don't think layerwise pretraining is tested using gradient sharing.

Aha! Link: https://skymindai.aha.io/features/DL4J-52

**Comment by lazydog2** [comment]
Before I submit a new issue about a question I have that seems related to this issue, I thought I'd ask my question here.
Can Spark be used to train a VAE given that the SparkDl4jMultiLayer class doesn't have a pretrain method? In case the answer differs between using parameter averaging and gradient sharing, I'm interested in the gradient sharing use case.

**Comment by agibsonccc** [comment]
@lazydog2 spark support will be discontinued in the next release. Not enough interest now a days. I'll mainly be focusing on single machine or multi gpu use cases. Closing this.

**Comment by lazydog2** [comment]
Thanks @agibsonccc, that's useful to know. I won't invest any further effort into using Deeplearning4j on Spark. I'd be interested in any other alternatives anyone knows about and may have also used.

**Comment by agibsonccc** [comment]
@lazydog2 nothing JVM based really works out of the box for distributed training. Most folks just train on multi gpu on a single box now. For large scale stuff something like ray or torch distributed is probably your best bet.
For most JVM workloads  if they do do any training it'll be on a single box for finetuning or something similar. You're better off running stuff on python and then importing the models. 

Unfortunately financially it doesn't make sense for java based frameworks to try to do more than that. If the users wanted it the money would have come from somewhere.
